src/pages/api/projects/nonprofit.ts

```diff
@@ -1,30 +1,59 @@
+import { JTDSchemaType } from "ajv/lib/types/jtd-schema";
 import {
   nonprofitCreateProject,
   nonprofitGetProjects,
 } from "server/mongodb/actions/Project";
-import APIWrapper from "server/utils/APIWrapper";
+import { APIWrapper } from "server/utils/APIWrapper";
+import { buildListQuerySchema } from "server/utils/request-validation";
 import {
+  InternalRequest,
   NonprofitCreateProject,
-  NonprofitGetProjects,
+  NonprofitProjectsListQuery,
+  ProjectType,
   Role,
 } from "src/utils/types";
 
+const GET_QUERY_SCHEMA: JTDSchemaType<NonprofitProjectsListQuery> =
+  buildListQuerySchema({
+    optionalProperties: {
+      status: {
+        optionalProperties: {
+          $active: {
+            type: "boolean",
+          },
+        },
+      },
+    },
+  });
+
+const POST_BODY_SCHEMA: JTDSchemaType<NonprofitCreateProject> = {
```

Comment:
We can infer types from schemas, however, I believe defining the type separately is a good way to "sanity-check" the schema.